// services/emailService.js
const nodemailer = require('nodemailer');

const transporter = nodemailer.createTransport({
  host:   process.env.SMTP_HOST,
  port:   parseInt(process.env.SMTP_PORT || '587'),
  secure: false,
  auth: {
    user: process.env.SMTP_USER,
    pass: process.env.SMTP_PASS,
  },
});

const sendVerificationEmail = async (to, token) => {
  const url = `${process.env.CLIENT_URL}/verify-email?token=${token}`;
  await transporter.sendMail({
    from: `"SwapMarket" <${process.env.SMTP_USER}>`,
    to,
    subject: 'Verify your SwapMarket account',
    html: `
      <h2>Welcome to SwapMarket!</h2>
      <p>Click the button below to verify your email address.</p>
      <a href="${url}" style="
        display:inline-block;padding:12px 24px;background:#1D9E75;
        color:#fff;border-radius:8px;text-decoration:none;font-weight:600
      ">Verify Email</a>
      <p style="color:#888;font-size:12px;margin-top:16px">
        Link expires in 24 hours. If you didn't sign up, ignore this email.
      </p>
    `,
  });
};

const sendPasswordReset = async (to, token) => {
  const url = `${process.env.CLIENT_URL}/reset-password?token=${token}`;
  await transporter.sendMail({
    from: `"SwapMarket" <${process.env.SMTP_USER}>`,
    to,
    subject: 'Reset your SwapMarket password',
    html: `
      <h2>Password Reset</h2>
      <p>Click below to reset your password. This link expires in 1 hour.</p>
      <a href="${url}" style="
        display:inline-block;padding:12px 24px;background:#534AB7;
        color:#fff;border-radius:8px;text-decoration:none;font-weight:600
      ">Reset Password</a>
    `,
  });
};

module.exports = { sendVerificationEmail, sendPasswordReset };
